### PR TITLE
PMA-020 (Edit profile)

### PR DESCRIPTION
**Pull request by allyksander**
I really wanted to create a good edit-profile route, but I lost...
Sorry((

### CODE REVIEW

File: src/constants/text.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@ export const registrationPageText = {
   successSignUp: 'was successfuly registered',
 };
 
+export const editProfilePageText = {
+  title: 'Edit Profile',
+  successSignUp: 'was successfuly udated',
```
Comment (AnAtoliy-AA) [inline src/constants/text.ts]: u`p`dated

File: src/pages/EditProfile.tsx [hunk 2]
```diff
@@ -1,7 +1,137 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Controller, useForm } from 'react-hook-form';
+import {
+  Alert, Box, Container, Avatar, Typography, TextField, Button,
```
Comment (AnAtoliy-AA) [inline src/pages/EditProfile.tsx]: write it in column, or in one string

File: src/pages/EditProfile.tsx [hunk 3]
```diff
@@ -1,7 +1,137 @@
...
+    // dispatch(getUserData(userId!));
+    // const { name, login } = userData;
+    // setValue(userAuthInput.properties.id as keyof SignUpFormInput, name);
+    // setValue(loginAuthInput.properties.id as keyof SignUpFormInput, login);
+    /* 
+    * TODO: I need to change this logic
+    */
+    UserService.getUserData(userId!).then(
+      (response) => {
+        setTimeout(() => {
```
Comment (AnAtoliy-AA) [inline src/pages/EditProfile.tsx]: why you used setTimeout

File: src/pages/EditProfile.tsx [hunk 4]
```diff
@@ -1,7 +1,137 @@
...
+          const { name, login } = response.data;
+
+          setValue(userAuthInput.properties.id as keyof SignUpFormInput, name);
+          setValue(loginAuthInput.properties.id as keyof SignUpFormInput, login);
+        });
+      },
+    );
+  }, []);
```
Comment (AnAtoliy-AA) [inline src/pages/EditProfile.tsx:46]: is this useEffect don't have dependencies?

File: src/pages/EditProfile.tsx [hunk 5]
```diff
@@ -1,7 +1,137 @@
...
+          {!isLoading && !newUser && (
+            <>
+              <RegistrationForm />
+              {error.message && (
+                <Alert sx={{ mb: '1rem' }} severity="error">
+                  {error.message}
+                </Alert>
+              )}
+            </>
+          )} */}
```
Comment (AnAtoliy-AA) [inline src/pages/EditProfile.tsx]: is this commented code will be usefull in future? if not - remove it

File: src/types/user.ts [hunk 6]
```diff
@@ -8,3 +10,14 @@ export interface NewUser {
...
+export type UserData = {
+  id: string;
+  name: string;
+  login: string;
+};
+
+export type UserDataParams = {
+  id: string;
+  userData: SignUpFormInput;
+}
```
Comment (AnAtoliy-AA) [inline src/types/user.ts]: empty line at the end of each file

File: src/pages/EditProfile.tsx [hunk 7]
```diff
@@ -1,7 +1,137 @@
...
+
+  useEffect(() => {
+    // dispatch(getUserData(userId!));
+    // const { name, login } = userData;
+    // setValue(userAuthInput.properties.id as keyof SignUpFormInput, name);
+    // setValue(loginAuthInput.properties.id as keyof SignUpFormInput, login);
+    /* 
+    * TODO: I need to change this logic
+    */
+    UserService.getUserData(userId!).then(
```
Comment (AnAtoliy-AA) [inline src/pages/EditProfile.tsx]: why you send requests from component??

File: src/constants/text.ts (hunk 1 above)
Comment (allyksander) [reply src/constants/text.ts]: Fixed.

File: src/pages/EditProfile.tsx (hunk 2 above)
Comment (allyksander) [reply src/pages/EditProfile.tsx]: May be was something with autoformating. Now I fixed it.

File: src/pages/EditProfile.tsx (hunk 3 above)
Comment (allyksander) [reply src/pages/EditProfile.tsx]: I fixed all this useEffect and replace requests out from component.

File: src/pages/EditProfile.tsx (hunk 4 above)
Comment (allyksander) [reply src/pages/EditProfile.tsx:46]: Fixed.

File: src/pages/EditProfile.tsx (hunk 5 above)
Comment (allyksander) [reply src/pages/EditProfile.tsx]: I removed it.

File: src/types/user.ts (hunk 6 above)
Comment (allyksander) [reply src/types/user.ts]: Fixed

File: src/pages/EditProfile.tsx (hunk 7 above)
Comment (allyksander) [reply src/pages/EditProfile.tsx]: I fixed it upper.

**Comment by Gibr** [comment]
I dint know why such error appear in thiis PR, but i have this:
Compiling error

solution:
Compiling error solution

**Comment by Gibr** [comment]
I think there is a few points todo, perhaps in other PR:
1. Some "success" text after profile edit done
2. Translate for fields

**Review by Gibr — COMMENTED** [review]
I think it is neccesary to fix one compile error, and then move on with next PR

**Pull request merged**